src/transport.rs

```diff
@@ -72,20 +73,30 @@ impl TransportProtocol {
 
 impl I2cTransport {
     pub fn new(path: &str, address: u16) -> Result<Self> {
+        let last_wake = Instant::now().checked_sub(Duration::from_secs(10)).unwrap();
         let mut port = I2c::from_path(path)?;
         port.smbus_set_slave_address(address, false)?;
 
-        Ok(Self { port, address })
+        Ok(Self { port, address, last_wake })
     }
 
-    fn send_wake(&mut self, wake_delay: Duration) -> Result {
-        let _ = self.send_buf(0, &[0]);
-        thread::sleep(wake_delay);
+    fn send_wake(&mut self, wake_delay: Duration, min_wake_interval: Duration) -> Result {
+        if Instant::now().checked_duration_since(self.last_wake) > Some(min_wake_interval) {
+            self.send_idle();
+            let _ = self.send_buf(0, &[0x00]);
+            thread::sleep(wake_delay);
+            self.last_wake = Instant::now();
```

Comment:
Heh so the question is whether you set last_wake _after_  or _before the thread::sleep